mmengine/dist/utils.py

```diff
@@ -101,10 +101,11 @@ def _init_dist_pytorch(backend, init_backend='torch', **kwargs) -> None:
     rank = int(os.environ['RANK'])
     if is_mlu_available():
         import torch_mlu  # noqa: F401
-        torch.mlu.set_device(rank)
+        local_rank = int(os.environ['LOCAL_RANK'])
+        torch.mlu.set_device(local_rank)
         torch_dist.init_process_group(
             backend='cncl',
-            rank=rank,
+            rank=local_rank,
```

Comment:
Should the rank be a number between 0 and world_size-1 rather than the local rank?

Reply:
Sorry for keeping you waiting. The rank in init_process_group is changed to global rank. The code is tested with `example/distributed_training_with_flexible_runner.py`. Now it can run resnet50 with multi-node properly.